database/models/measurements.js

```diff
@@ -10,8 +10,12 @@ module.exports = {
     create: function (data) {
         return databaseP.then(function (db) {
 
+        var cloneData = JSON.parse(JSON.stringify(data)); // Copy the object...
```

Comment:
Yep sounds good. Not to be done too often. More and more we should consider using immutable-js for more cases.